test/common/network/listen_socket_impl_test.cc

```diff
@@ -9,10 +9,18 @@ TEST(ListenSocket, All) {
   listen(socket1.fd(), 0);
   EXPECT_EQ(15000U, socket1.localAddress()->ip()->port());
 
-  EXPECT_THROW(Network::TcpListenSocket socket2(uint32_t(15000), true), EnvoyException);
+  TcpListenSocket socket2("tcp://127.0.0.1:15002", true);
+  listen(socket2.fd(), 0);
+  EXPECT_EQ("127.0.0.1:15002", socket2.localAddress()->asString());
 
-  TcpListenSocket socket2(dup(socket1.fd()), 15000);
-  EXPECT_EQ(15000U, socket2.localAddress()->ip()->port());
+  EXPECT_THROW(Network::TcpListenSocket socket3(uint32_t(15000), true), EnvoyException);
+  EXPECT_THROW(Network::TcpListenSocket socket4("tcp://127.0.0.1:15002", true), EnvoyException);
+
+  TcpListenSocket socket5(dup(socket1.fd()), 15000);
+  EXPECT_EQ(15000U, socket5.localAddress()->ip()->port());
+
+  TcpListenSocket socket6(dup(socket1.fd()), std::string("tcp://127.0.0.1:15004"));
```

Comment:
Can you comment on why we have this test as well as the socket2? In general, all these socket cases could do with a one line comment explaining the intent.